hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointManagerImpl.java

```diff
@@ -122,7 +122,7 @@ public void removeEndpoint(final ClientEndpoint ce, boolean closeImmediately) {
         final Connection connection = endpoint.getConnection();
         if (closeImmediately) {
             try {
-                connection.close();
+                connection.close(null, null);
```

Comment:
The client connection logic is messy and it doesn't provide value to add the cause, reason.

WHen a connection is closed, and event is send to the client engine which will then try to close the connection again. There is no point setting cause/reason on an already closed connection